src/ngraph/pass/constant_folding.cpp

```diff
@@ -85,3 +87,65 @@ void ngraph::pass::ConstantFolding::construct_constant_reshape()
     auto reshape_matcher = make_shared<pattern::Matcher>(reshape, constant_reshape_callback);
     this->add_matcher(reshape_matcher);
 }
+
+template <class T>
+shared_ptr<op::Constant> make_constant_broadcast(shared_ptr<op::Constant> constant,
+                                                 shared_ptr<op::Broadcast> broadcast)
+{
+    auto out_shape = broadcast->get_shape();
+    vector<T> out_vec(shape_size(out_shape));
+
+    runtime::reference::broadcast<T>(constant->get_vector<T>().data(),
+                                     out_vec.data(),
+                                     constant->get_shape(),
+                                     out_shape,
+                                     broadcast->get_broadcast_axes());
+
+    return make_shared<op::Constant>(constant->get_element_type(), out_shape, out_vec);
+}
+
+void ngraph::pass::ConstantFolding::construct_constant_broadcast()
+{
+    auto constant_label =
+        make_shared<pattern::op::Label>(element::f32, Shape{2}, pattern::has_class<op::Constant>());
+
+    auto broadcast = make_shared<op::Broadcast>(constant_label, Shape{2, 4}, AxisSet{1});
+
+    auto constant_broadcast_callback = [constant_label](pattern::Matcher& m) {
+        auto pattern_map = m.get_pattern_map();
+
```

Comment:
could you please add an `NGRAPH_DEBUG` message that we are inside a callback for this `construct_constant_broadcast`?

Reply:
DONE